### ISSUE TITLE
unable to open the database file during package installation

### ISSUE THREAD

**Issue by sanchom** [racket/racket#2691]
Trying to build an image using this dockerfile:

```
FROM ubuntu:19.04

RUN apt-get update && apt-get install -y \
  software-properties-common

RUN add-apt-repository ppa:plt/racket

RUN apt-get update && apt-get install -y \
  racket \
  openssl \
  sudo \
  ca-certificates \
  sqlite3

RUN raco pkg install --auto pollen hyphenate xexpr-path
```

During the final step (`rack pkg install --auto pollen hyphenate xexpr-path`), I get this error:

```
raco setup: docs failure: query-exec: unable to open the database file
  error code: 14
  SQL: "ATTACH $1 AS other"
  database: #<path:/root/.racket/7.3/doc/docindex.sqlite>
  mode: 'read-only
  file permissions: (write read)
```

**Comment by sanchom** [comment]
@takikawa Not sure if this is a PPA issue instead of a racket issue.

**Comment by ndw** [comment]
Seems like a Racket issue to me. If the racket document index doesn't exists, shouldn't racket create it?

**Comment by ndw** [comment]
On a hunch, I tried reworking my Dockerfile so that I wasn't attempting to install the packages as root. No change.

On a second hunch, I went back to Racket 7.0 and the problem went away (with the .sh installer rather than the Ubuntu installer).

**Comment by ndw** [comment]
Sorry, one last remark. It looks to be something in the Debian packaging. Downloading 7.4 and installing it from the .sh file worked fine too.

**Comment by jackfirth** [comment]
@ndw See also https://github.com/jackfirth/racket-docker, particularly the [dockerfile](https://github.com/jackfirth/racket-docker/blob/master/racket.Dockerfile).

**Comment by ndw** [comment]
That's downloading and using the .sh installer; that's consistent with what worked for me. Installing with `apt-get` appears to leave the database uninitialized.

**Comment by jackfirth** [comment]
Not sure if installing from the PPA in a docker container is supposed to work. Pinging @takikawa, who maintains the [Racket Ubuntu PPA][racket-ppa].

[racket-ppa]: https://github.com/takikawa/racket-ppa

**Comment by Immortalin** [comment]
I am still getting this issue on Ubuntu Focal @takikawa

**Comment by jasonhemann** [comment]
AFAIK, this is still outstanding.

**Comment by ed-flanagan** [comment]
TL;DR: I don't have an answer for the PPA issue, but I worked around the issue by executing

```
raco setup --doc-index --force-user-docs
```

before running `raco pkg install ...`.


Anyways, sorry for the long comment below. Hopefully the TL;DR work-around helps a little. The rest is just stuff I came across. Sharing on the off-chance something is useful. Thank you all for supporting the language!

---

Hey, I believe I'm running into a similar issue.
I'm just starting to try out Racket today (no experience in Racket, or Lisp in general), so apologies if I'm off in any way, or haven't provided a lot of depth, or mention something you already know.
Similar to @sanchom, I'm trying to create a Debian-based Dockerfile for Racket & the `sicp` package, utilizing APT directly (thank you @jackfirth for your docker link). While attempting to install the package using `raco`, I ran into a similar SQLite error on Debian Bullseye.
I'll spare the entire Dockerfile. However, after installing `racket` and `racket-doc`, I ran into the same error as @sanchom while executing `raco pkg install --auto --batch --fail-fast sicp`.

I checked out the intermediate image and `find / -type f -name "docindex.sqlite"` came up empty. 

I tried working-around by adding `mkdir -p /root/.local/share/racket/7.9/doc && touch /root/.local/share/racket/7.9/doc/docindex.sqlite` before the package install, but that just resulted in a different error. Presumably, `raco` expects that file to exist _and_ be populated.

```
#6 8.714 raco setup: --- building documentation ---                         [2:10:04]
#6 9.941 raco setup: docs failure: query-maybe-row: no such table: pathids
#6 9.941   error code: 1
#6 9.941 raco setup: --- installing collections ---                         [2:10:05]
#6 9.941 raco setup: --- post-installing collections ---                    [2:10:05]
#6 9.942 raco pkg install: packages installed, although setup reported errors
```

After poking around https://github.com/takikawa/racket-ppa (thanks again, @jackfirth), I gave up and just wondered how/when `docindex.sqlite` is generated. 

Guess-and-check, `raco setup --doc-index --force-user` seemed to do the trick. I kind of guessed `setup` from https://docs.racket-lang.org/pkg/implementation.html

That command seems to ensure `docindex.sqlite` is created in both
* `/usr/share/doc/racket/docindex.sqlite`
* `/root/.local/share/racket/7.9/doc/docindex.sqlite`

My best guess is `pkg install` might not try to create the `docindex.sqlite` file if it doesn't exist. These were some lines that I felt might be related, but honestly no clue.

* Where the `docs failure` message prefix is from: https://github.com/takikawa/racket-ppa/blob/7d98f7570c8bf23265738c36ddb201344cd3bfaf/collects/setup/setup-core.rkt#L1440-L1451
* An example where it expects the `/usr/share/...` and `.local/share/...` paths: https://github.com/takikawa/racket-ppa/blob/7d98f7570c8bf23265738c36ddb201344cd3bfaf/collects/setup/setup-core.rkt#L906-L911
* Some "db" paths: https://github.com/takikawa/racket-ppa/blob/45c09fe88d7d188103e4d3f1daa41b5689031886/collects/setup/private/pkg-deps.rkt#L424-L425
* Possible place to add a condition when the file doesn't exist?: https://github.com/takikawa/racket-ppa/blob/45c09fe88d7d188103e4d3f1daa41b5689031886/collects/setup/private/pkg-deps.rkt#L428
* Function that could be called for place above?: https://github.com/takikawa/racket-ppa/blob/7d98f7570c8bf23265738c36ddb201344cd3bfaf/share/pkgs/racket-index/setup/scribble.rkt#L244

Guess sounds similar to https://github.com/racket/racket/issues/2691#issuecomment-528524911

Maybe https://github.com/takikawa/racket-ppa/blob/45c09fe88d7d188103e4d3f1daa41b5689031886/collects/setup/private/pkg-deps.rkt#L428 can include https://github.com/takikawa/racket-ppa/blob/7d98f7570c8bf23265738c36ddb201344cd3bfaf/share/pkgs/racket-index/setup/scribble.rkt#L244 when `file-exists?` resolves to false (I really don't know Racket, so no clue if that's what's even going on)
